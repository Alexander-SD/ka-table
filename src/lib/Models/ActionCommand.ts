export class ActionCommand {
  private rejectedFlag: boolean = false;
  get rejected() {
    return this.rejectedFlag;
  }
  public reject(): void {
    this.rejectedFlag = true;
  }
}
