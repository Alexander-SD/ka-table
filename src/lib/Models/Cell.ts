/**
 * Describes the position of cell in table
 */
export class Cell {
  /*
  * Specifies the field of data which value will be used in cell
  */
  public field!: string;

  /*
  * Specifies the value of row's key where cell is placed
  */
  public rowKeyValue!: any;
}
