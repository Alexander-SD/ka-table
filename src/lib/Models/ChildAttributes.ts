import { ITableAllProps } from '../';
import { ICellContentProps } from '../Components/CellContent/CellContent';
import { IDataRowProps } from '../Components/DataRowContent/DataRowContent';
import { ChildAttributesItem } from '../types';

export class ChildAttributes {
  public cell?: ChildAttributesItem<ICellContentProps>;
  public table?: ChildAttributesItem<ITableAllProps>;
  public dataRow?: ChildAttributesItem<IDataRowProps>;
}
