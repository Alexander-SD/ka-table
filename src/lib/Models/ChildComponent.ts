import { PropsWithChildren } from 'react';

import { ChildAttributesItem } from '../types';

export class ChildComponent<TProps> {
  public elementAttributes?: ChildAttributesItem<TProps>;
  public propsAreEqual?: (prevProps: TProps, nextProps: TProps) => boolean;
  public content?: (props: PropsWithChildren<TProps>) => any
}