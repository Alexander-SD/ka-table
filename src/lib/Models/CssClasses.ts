export class CssClasses {
  public table?: string;
  public thead?: string;
  public checkbox?: string;
  public dateInput?: string;
  public numberInput?: string;
  public textInput?: string;
}
