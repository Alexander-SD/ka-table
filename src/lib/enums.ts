export enum DataType {
  Boolean = 'boolean',
  Date = 'date',
  Number = 'number',
  Object = 'object',
  String = 'string',
}

export enum EditingMode {
  None,
  Cell,
}

export enum KeyboardEnum {
  Esc = 27,
  Enter = 13,
}

export enum SortDirection {
  Ascend = 'ascend',
  Descend = 'descend',
}

export enum SortingMode {
  None,
  Single,
}

export enum TextAlign {
  Center = 'center',
  Left = 'left',
  Right = 'right',
}
