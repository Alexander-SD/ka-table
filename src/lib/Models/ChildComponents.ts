import { ITableProps } from '../';
import { ICellProps } from '../Components/CellComponent/CellComponent';
import { ICellContentProps } from '../Components/CellContent/CellContent';
import { ICellEditorProps, IFilterRowEditorProps } from '../Components/CellEditor/CellEditor';
import { IDataRowProps } from '../Components/DataRowContent/DataRowContent';
import { IGroupRowProps } from '../Components/GroupRowContent/GroupRowContent';
import { IHeadCellProps } from '../Components/HeadCell/HeadCell';
import { ChildComponent } from './ChildComponent';

export class ChildComponents {
  public cell?: ChildComponent<ICellProps>;
  public editor?: ChildComponent<ICellEditorProps>;
  public filterRowCell?: ChildComponent<IFilterRowEditorProps>;
  public groupCell?: ChildComponent<IGroupRowProps>;
  public headCell?: ChildComponent<IHeadCellProps>;
  public cellText?: ChildComponent<ICellContentProps>;
  public table?: ChildComponent<ITableProps>;
  public dataRow?: ChildComponent<IDataRowProps>;
  public detailsRow?: ChildComponent<IDataRowProps>;
}
