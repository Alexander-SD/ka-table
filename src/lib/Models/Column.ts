import { FilterCondition } from '../../../dist/Models/FilterCondition';
import defaultOptions from '../defaultOptions';
import { DataType, SortDirection } from '../enums';
import {
  CellFunc, EditorFunc, FormatFunc, HeaderCellFunc, SearchFunc, ValidationFunc,
} from '../types';

/**
 * Describes column of table its look and behaviour
 */
export class Column {
  public cell?: CellFunc;
  public dataType?: DataType = defaultOptions.columnDataType;
  public editor?: EditorFunc;
  public filterRowCell?: EditorFunc;
  public filterRowValue?: any;
  public filterRowOperator?: any;
  public field?: string;
  public format?: FormatFunc;
  public headCell?: HeaderCellFunc;
  public isEditable?: boolean;
  public key!: string;
  public search?: SearchFunc;
  public sortDirection?: SortDirection;
  public style?: React.CSSProperties;
  public title?: string;
  public validation?: ValidationFunc;
}
