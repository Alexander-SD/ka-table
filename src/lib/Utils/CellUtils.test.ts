import { EditingMode } from '../enums';
import { Cell } from '../Models/Cell';
import {
  changeCellEditorToCellTextHandler, changeCellTextToCellEditorHandler, isEditableCell,
} from './CellUtils';

describe('CellUtils', () => {
  it('isEditableCell equals true', () => {
    const rowEditableCells = isEditableCell(EditingMode.Cell, 'column', [{
      field: 'column',
      rowKeyValue: 10,
    }]);
    expect(rowEditableCells).toBeTruthy();
  });

  it('isEditableCell equals false', () => {
    const rowEditableCells = isEditableCell(EditingMode.Cell, 'column2', [{
      field: 'column',
      rowKeyValue: 10,
    }]);
    expect(rowEditableCells).toBeFalsy();
  });

  it('isEditableCell always equals false when EditingMode.None', () => {
    const rowEditableCells = isEditableCell(EditingMode.None, 'column', [{
      field: 'column',
      rowKeyValue: 10,
    }]);
    expect(rowEditableCells).toBeFalsy();
  });

  describe('CellHandlers', () => {
    it('changeCellTextToCellEditorHandler', () => {
      const onOptionChangedMock = jest.fn((x) => {});
      const editableCells: Cell[] = [{
        field: 'column',
        rowKeyValue: 1,
      }];
      const item: Cell = {
        field: 'column2',
        rowKeyValue: 2,
      };
      changeCellTextToCellEditorHandler(item, editableCells, onOptionChangedMock);
      expect(onOptionChangedMock.mock.calls.length).toBe(1);
      expect(onOptionChangedMock.mock.calls[0]).toMatchSnapshot();
    });

    it('changeCellEditorToCellTextHandler', () => {
      const onOptionChangedMock = jest.fn();
      const editableCells: Cell[] = [{
        field: 'column',
        rowKeyValue: 1,
      }, {
        field: 'column2',
        rowKeyValue: 2,
      }, {
        field: 'column',
        rowKeyValue: 2,
      }];
      const item: Cell = {
        field: 'column2',
        rowKeyValue: 2,
      };
      changeCellEditorToCellTextHandler(item, editableCells, onOptionChangedMock);
      expect(onOptionChangedMock.mock.calls.length).toBe(1);
      expect(onOptionChangedMock.mock.calls[0]).toMatchSnapshot();
    });
  });
});
