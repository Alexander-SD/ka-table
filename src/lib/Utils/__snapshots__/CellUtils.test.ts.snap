// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellUtils CellHandlers changeCellEditorToCellTextHandler 1`] = `
Array [
  Object {
    "editableCells": Array [
      Object {
        "field": "column",
        "rowKeyValue": 1,
      },
      Object {
        "field": "column",
        "rowKeyValue": 2,
      },
    ],
  },
]
`;

exports[`CellUtils CellHandlers changeCellTextToCellEditorHandler 1`] = `
Array [
  Object {
    "editableCells": Array [
      Object {
        "field": "column",
        "rowKeyValue": 1,
      },
      Object {
        "field": "column2",
        "rowKeyValue": 2,
      },
    ],
  },
]
`;
